src/transformers/__init__.py

```diff
@@ -406,9 +407,11 @@
 
 # TensorFlow
 if is_tf_available():
-    from .modeling_tf_utils import (
+    from .modeling_tf_generation import (
         shape_list,
```

Comment:
Why is `shape_list` in `modeling_tf_generation`?  I think we should leave it in `modeling_tf_utils` and import it. I think it's heavily used in other files as well

Reply:
We could import it dynamically if problems with circular imports arrive

Reply:
yeah I was having trouble with circular imports, will keep trying to make it work :)

Reply:
agree with @patrickvonplaten here. It's a very simple utility so don't fight too much with imports and copy it both in `modeling_tf_utils` and `modeling_tf_generation_utils` if you want (cf the [Rule of Three](https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)) that we follow in the lib, duplicating once is ok.